### ISSUE TITLE
Rusina ferruginea clssification failure 

### ISSUE THREAD

**Issue by larspett** [BiologicalRecordsCentre/ABLE#684]
Is Rusina ferruginea missing from the ButterflyCount species list in the Indicia data warehouse? ObsIdentify classifies it easily but ButterflyCount fails.I think I remember this happening before with other missing species?

![IMG_5602](https://github.com/BiologicalRecordsCentre/ABLE/assets/19598308/41400985-46ab-42a1-b32d-c78209c51a98)
![IMG_5601](https://github.com/BiologicalRecordsCentre/ABLE/assets/19598308/f7b7520c-7e10-47a7-a37f-ac4d020d8adf)
![processed-283B61A9-3E3F-4132-89CB-8B2767AE171E](https://github.com/BiologicalRecordsCentre/ABLE/assets/19598308/f7dcc266-a9b5-4678-9ea2-311fe8c19e86)

**Comment by kazlauskis** [comment]
You're right, the classifier identifies this as `Charanyca ferruginea` wiht 99% probability. I think this species is missing in the Indicia warehouse.

**Comment by larspett** [comment]
OK will you raise the issue with the Indicia warehouse people?

**Comment by larspett** [comment]
@karolis it is still not classified, can you look into this or pass me on to those that handle the lookup tables?

31 maj 2024 kl. 16:27 skrev Lars Pettersson ***@***.***>:

﻿
OK will you raise the issue with the Indicia warehouse people?

**Comment by larspett** [comment]
This is still happening for _Charanyca ferruginea_ and is also happening for _Korscheltellus (Hepialus) lupulinus_. Looks 100% like an lookup error since both species are common and the images are nice. @DavidRoy could you advise how to raise the issue with what Karolis (above) refers to as the Indicia warehouse people?

**Comment by larspett** [comment]
Hi, I am keen to move on with this one and the related [#542](https://github.com/BiologicalRecordsCentre/ABLE/issues/542) which seems to be on a hold? Can you help me passing them on to the correct people @CrisSevilleja ?

**Comment by larspett** [comment]
Right now we experience problems with 

- Rusina ferruginea
- Korscheltellus lupulinus
- Spilarctia lutea
- Smerinthus ocellata

**Comment by larspett** [comment]
All have similar synonyms, often with the ending -us instead of -a

**Comment by andrewvanbreda** [comment]
Hi @kazlauskis @larspett,

I have been assigned issue https://github.com/BiologicalRecordsCentre/ABLE/issues/783 which includes your issue in the list, so I will try my best to help.
(I also note issue 542 is also included in this list)

I am unfamiliar with the ButterflyCount app.
Are you able to confirm what species list ButterflyCount uses for moths on the Warehouse, so I can check the species list.

If it is the "EBMS Moths" species list I can tell you the following:

- Rusina ferruginea is already present, but Charanyca ferruginea is not currently present as a synonym
- Korscheltellus lupulinus is present but spelt as Korscheltellus lupulina and has no synonyms.
- Spilarctia lutea is already present and has no synonyms entered
- Smerinthus ocellata is already present and has no synonyms entered

All these entries were added to the warehouse 2020, so they are old.

I could add Charanyca ferruginea as a synonym of Rusina ferruginea if you can confirm that.

I also note you last comment.
Could there be a problem where the app/ObsIdentify has a list of names spelt with -us, and the Warehouse has -a (as noted with Korscheltellus lupulina above)
I am afraid I am unfamiliar with the app, so I am not sure where this problem would be occurring in the code (whether this comparison is failing in app code or on the Warehouse)
However, if this is the case, I would think it is the code that needs changing rather than the names by doing something like an external key/organism key comparison instead, and only using the name as a last resort if the keys are missing as there could be spelling variation there.

I hope some of those thoughts help in some way

**Comment by kazlauskis** [comment]
@andrewvanbreda the app uses this report to fetch the moth species `/index.php/services/rest/reports/projects/ebms/ebms_app_species_list.xml?taxon_list_id=260`. I don't know if it pulls in synonyms, but if it does, then adding Charanyca ferruginea as a synonym makes sense.

**Comment by andrewvanbreda** [comment]
Hi @kazlauskis,
Looks like that report just returns everything from a species list as long as Allow Data Entry is set.
I have added the synonym Charanyca ferruginea.

**Comment by larspett** [comment]
Hi, I was discussing this with @DavidRoy last week and maybe this is part of a larger question about how to harmonize with national synonyms. ,There are pleny of Swedish species with synonyms that are just sligthly different from the current data warehouse taxonomic backbone, in the case of Swedish Dyntaxa taxonomy there are many with the two last letters in the scientific species name differing between Swedish and General taxonomies. We could solve all those one after the other with specific tweaks to the synonym list but maybe there should be a more general approach to this. Adding Charanyca ferruginea is a good thing but I think we need to look at this across all the taxa and all countries to compile a match-mismatch matrix that we could deal with.

**Comment by larspett** [comment]
Like Smerinthus ocellata vs. Smerinthus ocellatus,

Spilarctia lutea vs.

- Spilosoma lutea
- Spilarctia luteum
- Spillarctia lutea
- Spilosoma luteum

etc

**Comment by andrewvanbreda** [comment]
Hi @larspett,

I think this issue might be worth re-discussing with David, and see if he wants to get John van Breda involved as I think there is a computing aspect to this also.

From a computer aspect alone, the only way I know of that would deal with this issue at the moment would be to include all variations of the names in the database as synonyms, but this may not be desirable. Also what if there is more than one way of writing the preferred name, I do not think we can currently support that.
Am sure there are problem other difficulties I haven't thought of.

I would consider raising this in a new Github issue, and perhaps close this issue as the original Charanyca ferruginea issue is now fixed (that problem was different as Charanyca ferruginea was previously completely missing)
